src/materials/H5MoltresMaterial.C

```diff
@@ -0,0 +1,480 @@
+                     _material_key + "/" + std::to_string(static_cast<int>(_XsTemperature[l])) +
+                         "/" + _file_map[xsec_names[j]]);
+      if (xsec_names[j] == "CHI_D" && !flag)
+      {
+        for (decltype(_num_groups) k = 1; k < _num_groups; ++k)
+          xsec_map["CHI_D"][k].push_back(0.0);
+        xsec_map["CHI_D"][0].push_back(1.0);
+        mooseWarning(
+            "CHI_D data missing -> assume delayed neutrons born in top group for material " +
+            _name);
+        continue;
+      }
+      if (!flag)
+        mooseError("Unable to open database " + _material_key + "/" +
+                   std::to_string(static_cast<int>(_XsTemperature[l])) + "/" +
+                   _file_map[xsec_names[j]]);
+
+      H5std_string DATASET_NAME(_material_key + "/" +
+                                std::to_string(static_cast<int>(_XsTemperature[l])) + "/" +
+                                _file_map[xsec_names[j]]);
+      dataset = file_open.openDataSet(DATASET_NAME);
+
+      H5::DataSpace filespace = dataset.getSpace();
+      int rank = filespace.getSimpleExtentNdims();
+      hsize_t dims[rank];
+      rank = filespace.getSimpleExtentDims(dims);
+      if (o != dims[0])
+        mooseError("Dimensions of " + _material_key + "/" + std::to_string(_XsTemperature[l]) +
+                   "/" + _file_map[xsec_names[j]] + " and num_groups do not match\n" +
+                   std::to_string(dims[0]) + "!=" + std::to_string(o));
+      double data_out[dims[0]];
+      dataset.read(data_out, H5::PredType::NATIVE_DOUBLE);
+
+      for (decltype(_num_groups) k = 0; k < o; ++k)
+      {
+        xsec_map[xsec_names[j]][k].push_back(data_out[k]);
+      }
+    }
+    switch (_interp_type)
+    {
+      case NONE:
+        for (decltype(_num_groups) k = 0; k < o; ++k)
+        {
+          _xsec_linear_interpolators[xsec_names[j]][k].setData(_XsTemperature,
+                                                               xsec_map[xsec_names[j]][k]);
+        }
+        break;
+      case LINEAR:
+        for (decltype(_num_groups) k = 0; k < o; ++k)
+        {
+          _xsec_linear_interpolators[xsec_names[j]][k].setData(_XsTemperature,
+                                                               xsec_map[xsec_names[j]][k]);
+        }
+        break;
+      case SPLINE:
+        for (decltype(_num_groups) k = 0; k < o; ++k)
+        {
+          _xsec_spline_interpolators[xsec_names[j]][k].setData(_XsTemperature,
+                                                               xsec_map[xsec_names[j]][k]);
+        }
+        break;
+      case MONOTONE_CUBIC:
+        for (decltype(_num_groups) k = 0; k < o; ++k)
+        {
+          _xsec_monotone_cubic_interpolators[xsec_names[j]][k].setData(_XsTemperature,
+                                                                       xsec_map[xsec_names[j]][k]);
+        }
+        break;
+    }
+  }
+}
+
+void
+H5MoltresMaterial::dummyComputeQpProperties()
```

Comment:
Yep, functions like this should not be repeated between MoltresMaterial and H5MoltresMaterial.